pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1815,6 +1815,29 @@ type APIServerConfig struct {
 	BearerToken string `json:"bearerToken,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=v1;V1;v2;V2
+type AlertmanagerAPIVersion string
+
+const (
+	AlertmanagerAPIVersion1       = AlertmanagerAPIVersion("V1")
+	AlertmanagerAPIVersion2       = AlertmanagerAPIVersion("V2")
+	AlertmanagerAPIVersionUnKnown = AlertmanagerAPIVersion("")
+)
+
+func (av *AlertmanagerAPIVersion) ToUpper() AlertmanagerAPIVersion {
+	if av == nil {
+		return AlertmanagerAPIVersionUnKnown
+	}
+
+	switch string(*av) {
```

Comment:
```suggestion
	switch strings.ToLower(string(*av)) {
```

So you don't have to compare with lower and upper case

wdyt?

Reply:
Thanks @nicolastakashi . Probably not needed to compare with lower and upper case.
```golang
        case "v1", "V1":
		return AlertmanagerAPIVersion1
	case "v2", "V2":
		return AlertmanagerAPIVersion2
```